CI: Add labels to issue and PR and improve output from submission wf

mickahell opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
-->

### Summary
Add labels to issue and PR from submission wf.
Improve the wf output.

### Details and comments
- [x] `submission` label to every submission issue and PR
- [x] `on hold` label to every submission issue and PR who doesn't pass the standard check
- [x]  remove `on hold` label if the submission passed
- [x] `ready` label to every submission issue and PR who passed the standard check
- [x] remove `ready` label if the submission doesn't passed anymore
- [x] Add a wf summary
- [x] Make the comment in the issue more clean
- [x] Make the comment in the PR more clean
- [x] Tests run
- [x] Clean

---
Closes #480

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/qiskit-community/ecosystem?pullRequest=481) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/qiskit-community/ecosystem?pullRequest=481) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: mickahell<br/>:x: github-actions<br/><hr/>**github-actions** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/qiskit-community/ecosystem?pullRequest=481) it.</sub>

mickahell commented:
@frankharkins ready for your review :)

I also implemented a new comment structured using the GitHub step summary. That's allow to simplify the code and have more clear output :)
<img width="932" alt="Capture d’écran 2023-08-06 à 17 17 20" src="https://github.com/qiskit-community/ecosystem/assets/20951376/892e7902-55da-49e2-8835-321b51f1c6ea">

All the examples :
- issue passed : https://github.com/mickahell/ecosystem/issues/161
- run summary : https://github.com/mickahell/ecosystem/actions/runs/5777274128
- issue failed : https://github.com/mickahell/ecosystem/issues/160
- run summary : https://github.com/mickahell/ecosystem/actions/runs/5777271007

(the 2 errors in the annotations are normal it's because I'm asking to delete labels who aren't on the issue, I added a `continue-on-error: true` in order to ignore them)

mickahell commented:
@frankharkins seems the CLA licence is blocked because I added code from GitHub actions. 
I clean this code in my last commit (https://github.com/qiskit-community/ecosystem/pull/481/commits/9a72179f0ff1af8b8c573d6b93819ce491888556), so they aren't any code from GitHub actions anymore but the CLA seems to still not getting it.

mickahell commented:
The long line of commits was because I had to amend a commit from GitHub action ^^'

frankharkins commented on .github/workflows/ecosystem-submission.yml:
```diff
@@ -125,63 +100,167 @@ jobs:
+            repo: context.repo.repo,
+            labels: ["on hold"]
+          })
+    - name: PR labeling status on hold remove ready
+      continue-on-error: true
+      uses: actions/github-script@v6
+      if: ${{ steps.check-return.outputs.PASS_STD != 'True' }}
+      with:
+        script: |
+          github.rest.issues.removeLabel({
+            issue_number: context.issue.number,
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            name: ["ready"]
+          })
```
Is this duplicate of 194-205?

mickahell replied:
Not here is for the issue, on line 170 is for the PR

frankharkins commented on .github/workflows/ecosystem-submission.yml:
```diff
@@ -125,63 +100,167 @@ jobs:
 
-    - name: Create comment on failure for dev check
-      if: ${{ steps.check-return.outputs.PASS_DEV != 'True' }}
+    - name: Issue create comment
```
```suggestion
    - name: 'Issue: Post comment'
```

frankharkins commented on .github/workflows/ecosystem-submission.yml:
```diff
@@ -125,63 +100,167 @@ jobs:
-    - name: Create comment on success for dev check
-      if: ${{ steps.check-return.outputs.PASS_DEV == 'True' }}
-      uses: peter-evans/create-or-update-comment@v3
+    - name: Issue labeling status on hold
```
```suggestion
    - name: 'Issue: Add "on hold" label'
```

frankharkins reviewed (changes requested):
Look great, thanks!

Only problem is I think a couple of steps are duplicated. I've also suggested some name / comment changes to make things clearer at a glance.

mickahell commented on .github/workflows/ecosystem-submission.yml:
```diff
@@ -125,63 +100,167 @@ jobs:
+            repo: context.repo.repo,
+            labels: ["ready"]
+          })
+    - name: PR labeling status ready remove on hold
+      continue-on-error: true
+      uses: actions/github-script@v6
+      if: ${{ steps.check-return.outputs.PASS_STD == 'True' }}
+      with:
+        script: |
+          github.rest.issues.removeLabel({
+            issue_number: context.issue.number,
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            name: ["on hold"]
+          })
```
Not here is for the issue, on line 170 is for the PR

mickahell commented:
> Look great, thanks!
> 
> Only problem is I think a couple of steps are duplicated. I've also suggested some name / comment changes to make things clearer at a glance.

Yep @frankharkins I took your feedbacks :)
The duplicated aren't, they are about PR or issue the blocks are almost the same except the variable for the _issue_ id

frankharkins reviewed (changes requested):
Sorry to request more changes. I now understand there's a separate step for each label addition/removal; it might be clearer to combine these eight steps into two steps.

That is, this pattern

```
if condition:
    do x
    do y
    do z
```

rather than this one

```
if condition
    do x
if condition
    do y
if condition
    do z
```

Then there should be less code and it should be harder to miss something when modifying the action.

mickahell commented:
Hmm ok I see what you mean, I'm not sure I can merge stps together. I'l check that

frankharkins commented:
Can you just put all the code in the same script? E.g.

```yaml
with:
  script: |
    github.rest.issues.addLabel({
      ...
    })
    github.rest.issues.removeLabel({
      ...
    })
```

mickahell commented:
> Can you just put all the code in the same script? E.g.

Totally ! I was somewhere else and totally forget I can merge script together ^^'

> Sorry to request more changes. I now understand there's a separate step for each label addition/removal; it might be clearer to combine these eight steps into two steps.

Don't worry that's totally ok :)

Done, I mixed them. You can review it

frankharkins reviewed (approved):
Looks good, thank you!
